src/coordinates.h

```diff
@@ -7,6 +7,177 @@
 #include "enums.h"
 #include "game_constants.h"
 #include "point.h"
+#include "debug.h"
+
+namespace coords
+{
+
+enum class scale {
```

Comment:
Maybe a tag *type* (not enumeration value) is better. It allows easier adding of new systems. All properties (like scaling/absolute vs relative) can be done withing that tag type.

E.g.
```C++
class sub_map_system;

class map_system {
// this would allow converting map_system to sub_map_system, but not to say vehicle coordinates
// instead of a debug message this would yield a compiler error (as there is no factor_to( vehicle_system) function)
    static int factor_to( const sub_map_system & ) {
        return 12;
    }
};
```

Reply:
I think the code is simpler with two independent features rather than a tag type that controls all aspects.  Could use types rather than an enum.  I chose the enum because more meta-programming logic can be written in "regular C++" `constexpr` functions, rather than type-specialization style, which I was hoping would be clearer and easier to follow.

I'll try to keep both approaches in mind, though, if this version gets unwieldy.